prow/deck/jobs/jobs.go

```diff
@@ -141,7 +141,60 @@ func (ja *JobAgent) GetJobLog(job, id string) ([]byte, error) {
 	}
 	ja.mut.Unlock()
 	if !ok {
-		return nil, fmt.Errorf("no such job found: %s (id: %s)", job, id)
+		return kube.ProwJob{}, fmt.Errorf("no such job found: %s (id: %s)", job, id)
+	}
+	return j, nil
+}
+
+// GetJobLogTail returns the last n bytes of the job logs, works for both kubernetes and jenkins agent types.
+func (ja *JobAgent) GetJobLogTail(job, id string, n int64) ([]byte, error) {
+	j, err := ja.getProwJob(job, id)
+	if err != nil {
+		logrus.WithError(err).Error("Error getting ProwJob.")
+	}
+	if j.Spec.Agent == kube.KubernetesAgent {
+		client, ok := ja.pkcs[j.ClusterAlias()]
+		if !ok {
+			return nil, fmt.Errorf("cannot get logs for prowjob %q with agent %q: unknown cluster alias %q", j.ObjectMeta.Name, j.Spec.Agent, j.ClusterAlias())
+		}
+		return client.GetLogTail(j.Status.PodName, kube.TestContainerName, n)
+	}
+	for _, agentToTmpl := range ja.c.Config().Deck.ExternalAgentLogs {
+		if agentToTmpl.Agent != string(j.Spec.Agent) {
+			continue
+		}
+		if !agentToTmpl.Selector.Matches(labels.Set(j.ObjectMeta.Labels)) {
+			continue
+		}
+		var b bytes.Buffer
+		if err := agentToTmpl.URLTemplate.Execute(&b, &j); err != nil {
+			return nil, fmt.Errorf("cannot execute URL template for prowjob %q with agent %q: %v", j.ObjectMeta.Name, j.Spec.Agent, err)
+		}
+		resp, err := http.Get(b.String())
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+		content, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			logrus.WithError(err).Error("Error reading response body")
+		}
+		cr := bytes.NewReader(content)
```

Comment:
What does `cr` stand for?

Reply:
content reader

Reply:
(don't use abbreviations for variable names)

Reply:
This entire package is chock full of abbreviated variable names, and ones that are FAR more widely used than this one-line reader. (`ja`, `pkcs`, `mut`, `c`, `j`, `pjs`, `njs`, just to name a few) If we are going to commit to that style, it should be addressed in a different issue, and should definitely prioritize names like the ones I mentioned over something as trivial as this reader. Further, I think it's reasonable to allow some of the importance of a variable to be conveyed in the verbosity of its name.

Reply:
I probably wouldn't block this PR on that name, but FWIW, I'm not a fan of these. abbreviated variables names are common in go for:
- iteration (i,j,k)
- receiver type for methods on types `func (mt *MyType) Foo()`

Which both seem reasonable.. the latter is basically `self`... Otherwise I'm generally not a fan. go generally should use names more like `contentReader` which is a bit more obvious without as much context.

I would be in favor of any PRs that went back through prow and gave these more meaningful names than `njs`, `pjs` etc..

Reply:
So is this just accepted as a golang design mistake or? It seems like the name of this variable is pretty consistent with this claimed best practice. https://github.com/golang/go/wiki/CodeReviewComments#variable-names

Reply:
Quoting from the end of that section
> The basic rule: the further from its declaration that a name is used, the more descriptive the name must be. For a method receiver, one or two letters is sufficient. Common variables such as loop indices and readers can be a single letter (i, r). More unusual things and global variables need more descriptive names.

Per this, a reader might be OK to name `r`, but in many cases short names are less acceptable, especially those that are less common or have wider scope.

Reply:
So from my understanding then, this reader variable is used exactly once, 2 lines away from its declaration. It would therefore be acceptable for me to call it `r`. Calling it `cr` then also seems entirely reasonable. The other variables in this package absolutely should be renamed, but that is not within the scope of this PR.